src/providers/goDaddy.ts

```diff
@@ -1,56 +1,44 @@
-/*eslint @typescript-eslint/camelcase: 0*/
+/* eslint @typescript-eslint/camelcase: 0 */
+/* eslint @typescript-eslint/no-explicit-any: 0 */
+
 import { sendRequest } from '../helpers/httpRequest'
 import { getProviderKeys } from '../api/lib/data'
 import { Provider } from '../api/types/general'
+import { ServiceKey } from '../api/types/admin'
 
-export const getDomains = async (): Promise<Provider> => {
-  const service = 'https://api.godaddy.com'
-  const name = 'Godaddy'
-  const serviceKeys = getProviderKeys()
+const name = 'Godaddy'
+const service = 'https://api.godaddy.com'
 
-  const defaultKey = { value: '' }
-  const GD_Key = (serviceKeys.find(el => el.key === 'GD_Key') || defaultKey)
-    .value
-  const GD_Secret = (
-    serviceKeys.find(el => el.key === 'GD_Secret') || defaultKey
-  ).value
+const getKeys = (): ServiceKey[] => {
+  const keysDefault: { key: string }[] = [
+    { key: 'GD_Key' },
+    { key: 'GD_Secret' }
+  ]
+  const keys = keysDefault.map(keyInfo => {
+    const serviceKeys = getProviderKeys()
+    return (
+      serviceKeys.find(k => k.service === 'dns_gd' && k.key === keyInfo.key) ||
+      keysDefault
+    )
+  })
+  return keys as ServiceKey[]
+}
+const findKey = (key: string): string => {
+  return getKeys().find(k => k.key === key).value
```

Comment:
I got a TS warning `Object is possibly 'undefined'.ts(2532)`
I don't get it because I typed everything